composer/utils/object_store/gs_object_store.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Google Cloud SDK - Compatible object store."""
+
+from __future__ import annotations
+
+import os
+import pathlib
+import uuid
+from typing import Callable, Optional, Union
+
+from composer.utils.import_helpers import MissingConditionalImportError
+from composer.utils.object_store.object_store import ObjectStore
+
+__all__ = ['GsObjectStore']
+
+BOTOCORE_CLIENT_ERROR_CODES = ('403', '404', 'NoSuchKey')
+
+
+def _reraise_gs_errors(uri: str, e: Exception):
+    try:
+        from google.api_core.exceptions import GatewayTimeout, NotFound
+
+        print('Reraising exception: {e.message}')
```

Comment:
we shouldn't use print anywhere. should always use the logging library. please see other uses for examples.